Check/Users/RolesList.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * @file
+ * Contains \SiteAudit\Check\Users\RolesList.
+ */
+
+/**
+ * Class SiteAuditCheckUsersRolesList.
+ */
+class SiteAuditCheckUsersRolesList extends SiteAuditCheckAbstract {
+  /**
+   * Implements \SiteAudit\Check\Abstract\getLabel().
+   */
+  public function getLabel() {
+    return dt('List Roles');
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getDescription().
+   */
+  public function getDescription() {
+    return dt('Show all available roles and user counts.');
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultFail().
+   */
+  public function getResultFail() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultInfo().
+   */
+  public function getResultInfo() {
+    $counts = array();
+    foreach ($this->registry['roles'] as $name => $count_users) {
+      $counts[] = "$name: $count_users";
+    }
+    return implode(', ', $counts);
+  }
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultPass().
+   */
+  public function getResultPass() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getResultWarn().
+   */
+  public function getResultWarn() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\getAction().
+   */
+  public function getAction() {}
+
+  /**
+   * Implements \SiteAudit\Check\Abstract\calculateScore().
+   */
+  public function calculateScore() {
+    $roles = array_keys(\Drupal::entityManager()->getListBuilder('user_role')->load());
+    foreach ($roles as $role) {
```

Comment:
In drupal 8, there is no separate table for roles. Roles are stored as configuration in the `config` table along with all the other config entities of the drupal 8 site in the form of a string - `user.role.[role-name]`. The role in the `user__roles` table is stored just as `[role-name]`. I could not find a way to join the two tables. That is why I had to resort to this method.